CMakeLists.txt

```diff
@@ -740,7 +739,7 @@ if(SYSTEM_LIBS)
 endif()
 if(PNG_FOUND)
     include_directories(SYSTEM ${PNG_INCLUDE_DIRS})
-    target_link_libraries(${SHORT_NAME} ${PNG_LIBRARIES})
+    target_link_libraries(${SHORT_NAME} PRIVATE ${PNG_LIBRARIES})
```

Comment:
Apparently the Vita build and CodeQL don't like this. Any particular reason you added it?

Reply:
When I added the target_link_libraries for Switch I set it to PRIVATE then got an error about mixing signatures as per https://cmake.org/cmake/help/latest/policy/CMP0023.html

We can either set all of these to PRIVATE or just leave them unlabelled. I think the latter is probably the least intrusive here but it might be worth thinking about.

Reply:
Since Julius is an executable and not a library, there's nothing to be propagated. But we should indeed consider making everything `PRIVATE` at some point. For now, leaving it unlabelled is indeed fine.